src/DataProtection/DataProtection/src/KeyManagement/XmlKeyManager.cs

```diff
@@ -500,6 +501,8 @@ void IInternalXmlKeyManager.RevokeSingleKey(Guid keyId, DateTimeOffset revocatio
                 {
                     if (OSVersionUtil.IsWindows())
                     {
+                        Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Windows)); // Hint for the platform compatibility analyzer.
```

Comment:
I realize this helps the analyzer but why not use `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)` in the conditions instead of `OSVersionUtil.IsWindows()`❔

Reply:
OSVerstionUtil checks if the OS is Win7 or newer. I suspect the platform check is a direct replacement, but I didn't want to touch crypto code

Reply:
Makes sense

Reply:
If you were able to use `OperatingSystem.IsWindowsVersionAtLeast()`, then you'd be able to get a cleaner replacement, but that would require using that `net5.0` API, which I presume can't be depended upon here. Just something to consider in 6.0, but what you have here makes the most sense for 5.0.